pkg/kubelet/kubelet.go

```diff
@@ -715,6 +715,11 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 
 // registerWithApiserver registers the node with the cluster master.
 func (kl *Kubelet) registerWithApiserver() {
+	if _, err := kl.kubeClient.Nodes().Get(kl.hostname); err == nil {
```

Comment:
Seems like a bad idea-- what if the external ID changed? What is the harm of going through the loop below?